src/autoupdate.c

```diff
@@ -105,11 +127,20 @@ enum swupd_code autoupdate_main(int argc, char **argv)
 		int rc;
 		check_root();
 		info("Running systemctl to enable updates\n");
-		rc = systemctl_cmd("unmask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
+		rc = systemctl_cmd_path(root_path,
+					"unmask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
 		if (rc != 0) {
 			return SWUPD_SUBPROCESS_ERROR;
 		}
-		rc = systemctl_restart("swupd-update.timer");
+
+		/* If root_path is given and we have succeeded so far
+		 * make sense to return SWUPD_OK at this point
+		 */
+		if (root_path) {
+			return SWUPD_OK;
```

Comment:
So if we provide root_option, we really cant restart a service because it needs to work with the systemd daemon there. 

So the only thing to be done when we provide root_option is to mask and unmask files.

Reply:
ok, got it, thanks for the clarification, maybe adding that comment there to explain that in the code for future reference?

Reply:
Done